rdkit-stubs/Chem/Descriptors.pyi

```diff
@@ -19,31 +20,104 @@ _descList: Incomplete
-def MinPartialCharge(mol, force: bool = ...): ...
-def MaxAbsPartialCharge(mol, force: bool = ...): ...
-def MinAbsPartialCharge(mol, force: bool = ...): ...
-def _FingerprintDensity(mol, func, *args, **kwargs): ...
-def FpDensityMorgan1(x): ...
-def FpDensityMorgan2(x): ...
-def FpDensityMorgan3(x): ...
+>>> ExactMolWt(Chem.MolFromSmiles('CC'))
+30.04...
+>>> ExactMolWt(Chem.MolFromSmiles('[13CH3]C'))
+31.05...
+"""
+def NumValenceElectrons(mol: Chem.Mol) -> int:
+    """ 
+    The number of valence electrons the molecule has
 
-names: Incomplete
-autocorr: Incomplete
+    >>> NumValenceElectrons(Chem.MolFromSmiles('CC'))
+    14
+    >>> NumValenceElectrons(Chem.MolFromSmiles('C(=O)O'))
+    18
+    >>> NumValenceElectrons(Chem.MolFromSmiles('C(=O)[O-]'))
+    18
+    >>> NumValenceElectrons(Chem.MolFromSmiles('C(=O)'))
+    12
+    """
+def NumRadicalElectrons(mol: Chem.Mol) -> int:
+    """ 
+    The number of radical electrons the molecule has
+    (says nothing about spin state)
 
-def setupAUTOCorrDescriptors() -> None: ...
+    >>> NumRadicalElectrons(Chem.MolFromSmiles('CC'))
+    0
+    >>> NumRadicalElectrons(Chem.MolFromSmiles('C[CH3]'))
+    0
+    >>> NumRadicalElectrons(Chem.MolFromSmiles('C[CH2]'))
+    1
+    >>> NumRadicalElectrons(Chem.MolFromSmiles('C[CH]'))
+    2
+    >>> NumRadicalElectrons(Chem.MolFromSmiles('C[C]'))
+    3
+    """
+def _ChargeDescriptors(mol: Chem.Mol, force: bool = False) -> tuple[float, float]: ...
+def MaxPartialCharge(mol: Chem.Mol, force: bool = False) -> float: ...
+def MinPartialCharge(mol: Chem.Mol, force: bool = False) -> float: ...
+def MaxAbsPartialCharge(mol: Chem.Mol, force: bool = False) -> float: ...
+def MinAbsPartialCharge(mol: Chem.Mol, force: bool = False) -> float: ...
+def _FingerprintDensity(mol: Chem.Mol, func, *args, **kwargs) -> float: ...
+def FpDensityMorgan1(x) -> float: ...
+def FpDensityMorgan2(x) -> float: ...
+def FpDensityMorgan3(x) -> float: ...
+
+def setupAUTOCorrDescriptors() -> None:
+    """Adds AUTOCORR descriptors to the default descriptor lists"""
 
 class PropertyFunctor(rdMolDescriptors.PythonPropertyFunctor):
+    """
+    Creates a python based property function that can be added to the
+    global property list.  To use, subclass this class and override the
+    __call__ method.  Then create an instance and add it to the
+    registry.  The __call__ method should return a numeric value.
+
+    Example:
+
+        class NumAtoms(Descriptors.PropertyFunctor):
+            def __init__(self):
+                Descriptors.PropertyFunctor.__init__(self, "NumAtoms", "1.0.0")
+            def __call__(self, mol):
+                return mol.GetNumAtoms()
+
+        numAtoms = NumAtoms()
+        rdMolDescriptors.Properties.RegisterProperty(numAtoms)
+    """
     def __init__(self, name, version) -> None: ...
     def __call__(self, mol) -> Any: ...
 
+def CalcMolDescriptors(mol: Chem.Mol, missingVal: Optional[float] = None, silent: bool = True):
```

Comment:
```suggestion
def CalcMolDescriptors(mol: Chem.Mol, missingVal: Optional[float] = None, silent: bool = True) -> str[str, float]:
```

Reply:
it should be `dict[str, float | int]`. Thank you for your suggestion. I have oversighted.